File: 2634-minimum-common-value/2634-minimum-common-value.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        if nums1[0] > nums2[len(nums2) - 1] or nums2[0] > nums1[len(nums1) - 1] :
            return -1
        i =j = 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] < nums2[j]:
                i += 1
            elif nums1[i] > nums2[j]:
                j+=1
            else:
                return nums1[i]
        return -1

File: 2634-minimum-common-value/2634-minimum-common-value.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        if(nums1[0] > nums2[nums2.length - 1] || nums2[0] > nums1[nums1.length - 1])
            return -1;
        int i = 0, j = 0;
        while (i < nums1.length && j < nums2.length){
            if (nums1[i] > nums2[j]){
                j++;
            }
            else if(nums2[j] > nums1[i])
                i++;
            else{
                return nums1[i];
            }
        }
        return -1;
    }
}